dev/com.ibm.ws.microprofile.config.1.4.fragment/src/com/ibm/ws/microprofile/config14/fragment/Config14ProviderResolverOSGiStub.java

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.config14.fragment;
+
+import javax.management.ServiceNotFoundException;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.spi.ConfigBuilder;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+
+import com.ibm.wsspi.kernel.service.utils.FrameworkState;
+
+public class Config14ProviderResolverOSGiStub extends ConfigProviderResolver {
+
+    /** {@inheritDoc} */
+    @Override
+    public Config getConfig() {
+        return getConfigProviderResolver().getConfig();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Config getConfig(ClassLoader loader) {
+        return getConfigProviderResolver().getConfig(loader);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigBuilder getBuilder() {
+        return getConfigProviderResolver().getBuilder();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void registerConfig(Config config, ClassLoader classLoader) {
+        getConfigProviderResolver().registerConfig(config, classLoader);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void releaseConfig(Config config) {
+        getConfigProviderResolver().releaseConfig(config);
+    }
+
+    /**
+     * Get the OSGi bundle context for the given class
+     *
+     * @param clazz the class to find the context for
+     * @return the bundle context
+     */
+    static BundleContext getBundleContext() {
+        BundleContext context = null; //we'll return null if not running inside an OSGi framework (e.g. unit test) or framework is shutting down
```

Comment:
Probably better not to do this.

Elsewhere in this class, you assume that the result of `getBundleContext()` is not null.